crates/blockchain-tree/src/noop.rs

```
@@ -0,0 +1,133 @@
+
+impl BlockchainTreeEngine for NoopBlockchainTree {
+    fn buffer_block(&self, _block: SealedBlockWithSenders) -> Result<(), InsertBlockError> {
+        Ok(())
+    }
+
+    fn insert_block(
+        &self,
+        block: SealedBlockWithSenders,
+    ) -> Result<InsertPayloadOk, InsertBlockError> {
+        Err(InsertBlockError::tree_error(
+            BlockchainTreeError::BlockHashNotFoundInChain { block_hash: block.hash },
+            block.block,
+        ))
+    }
+
+    fn finalize_block(&self, _finalized_block: BlockNumber) {}
+
+    fn connect_buffered_blocks_to_canonical_hashes_and_finalize(
+        &self,
+        _last_finalized_block: BlockNumber,
+    ) -> RethResult<()> {
+        Ok(())
+    }
+
+    fn connect_buffered_blocks_to_canonical_hashes(&self) -> RethResult<()> {
+        Ok(())
+    }
+
+    fn make_canonical(&self, block_hash: &BlockHash) -> RethResult<CanonicalOutcome> {
+        Err(BlockchainTreeError::BlockHashNotFoundInChain { block_hash: *block_hash }.into())
+    }
+
+    fn unwind(&self, _unwind_to: BlockNumber) -> RethResult<()> {
+        Ok(())
+    }
+}
+
+impl BlockchainTreeViewer for NoopBlockchainTree {
+    fn blocks(&self) -> BTreeMap<BlockNumber, HashSet<BlockHash>> {
+        Default::default()
+    }
+
+    fn header_by_hash(&self, _hash: BlockHash) -> Option<SealedHeader> {
+        None
+    }
+
+    fn block_by_hash(&self, _hash: BlockHash) -> Option<SealedBlock> {
+        None
+    }
+
+    fn buffered_block_by_hash(&self, _block_hash: BlockHash) -> Option<SealedBlock> {
+        None
+    }
+
+    fn buffered_header_by_hash(&self, _block_hash: BlockHash) -> Option<SealedHeader> {
+        None
+    }
+
+    fn canonical_blocks(&self) -> BTreeMap<BlockNumber, BlockHash> {
+        Default::default()
+    }
+
+    fn find_canonical_ancestor(&self, _parent_hash: BlockHash) -> Option<BlockHash> {
+        None
+    }
+
+    fn is_canonical(&self, _hash: BlockHash) -> RethResult<bool> {
+        Ok(false)
```

should we return an error here?

---

yeah good point